decider-bcl2fastq/src/main/java/ca/on/oicr/pde/deciders/utils/BarcodeComparison.java

```diff
@@ -0,0 +1,82 @@
+package ca.on.oicr.pde.deciders.utils;
+
+import ca.on.oicr.pde.deciders.data.Barcode;
+import ca.on.oicr.pde.deciders.data.BarcodeCollision;
+import com.google.common.base.Strings;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.commons.text.similarity.HammingDistance;
+import org.apache.commons.text.similarity.LevenshteinDistance;
+
+/**
+ *
+ * @author mlaszloffy
+ */
+public class BarcodeComparison {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add a private constructor to hide the implicit public one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1118)